lib/services/access_checker.go

```diff
@@ -832,6 +836,7 @@ func (a *accessChecker) CheckAccessToRemoteCluster(rc types.RemoteCluster) error
 			return trace.Wrap(err)
 		}
 		if matchLabels {
+			setRoleAsUsed(role)
```

Comment:
Take into consideration that one can also have roles that by this definition aren't used - i.e. a role that enforces moderated sessions or trusted devices doesn't need to match the resource labels  - but in reality they enforce one of the most fundamental enforcements and might be highly misleading to say they aren't used at all